pkg/agent/agent.go

```diff
@@ -89,9 +91,47 @@ func (a *Agent) Run(ctx context.Context) error {
 		nodeAttestor = cat.GetNodeAttestor()
 	}
 
-	as, err := a.attest(ctx, sto, cat, metrics, nodeAttestor)
-	if err != nil {
-		return err
+	var as *node_attestor.AttestationResult
+	if a.c.RetryBootstrap {
+		attBackoffClock := clock.New()
+		attBackoff := backoff.NewBackoff(attBackoffClock, 5*time.Second, backoff.WithMaxElapsedTime(1*time.Minute))
+
+		for {
+			select {
+			case <-ctx.Done():
+				return ctx.Err()
+			default:
+			}
+
+			as, err = a.attest(ctx, sto, cat, metrics, nodeAttestor)
+			if err == nil {
+				break
+			}
+
+			switch status.Convert(err).Code() {
+			//TODO: what additional errors we do need here?
+			// https://github.com/grpc/grpc/blob/master/doc/statuscodes.md
+			case codes.InvalidArgument:
+				return err
+			}
+
+			nextDuration := attBackoff.NextBackOff()
+			if nextDuration == backoff.Stop {
+				return err
+			}
+
```

Comment:
We need to inform that there was an error. Something like this may work:
```
			a.c.Log.WithFields(logrus.Fields{
				telemetry.Error:         err,
				telemetry.RetryInterval: nextDuration,
			}).Warn("Attestation failed")

```